addons/project/models/project.py

```diff
@@ -1700,6 +1708,68 @@ def _compute_display_parent_task_button(self):
         for task in self:
             task.display_parent_task_button = task.parent_id in accessible_parent_tasks
 
+    def get_group_pattern(self):
+        return {
+            'tags_and_users': r'\s([#@]%s[^\s]+)',
+            'priority': r'\s(!)$',
+        }
+
+    def get_groups_patterns(self):
+        group_pattern = self.get_group_pattern()
+        return [
+            r'(?:%s)*' % (group_pattern['tags_and_users'] % ''),
+            r'(?:%s)?' % group_pattern['priority'],
+        ]
+
+    def get_cannot_start_with_patterns(self):
+        return [r'(?![#!@\s])']
+
+    def extract_tags_and_users(self):
+        tags = []
+        users = []
+        tags_and_users_group = self.get_group_pattern()['tags_and_users']
+        for word in re.findall(tags_and_users_group % '', self.display_name):
+            (tags if word.startswith('#') else users).append(word[1:])
+        users_to_keep = []
+        user_ids = []
+        for user in users:
+            matched_users = self.env['res.users'].name_search(user)
+            if len(matched_users) == 1:
+                user_ids.append(Command.link(matched_users[0][0]))
+            else:
+                users_to_keep.append(r'%s\b' % user)
+        self.user_ids = user_ids
+        if tags:
+            domain = expression.OR([[('name', '=ilike', tag)] for tag in tags])
+            existing_tags = self.env['project.tags'].search(domain)
+            existing_tags_names = {tag.name.lower() for tag in existing_tags}
+            new_tags_names = set(tag for tag in tags if tag.lower() not in existing_tags_names)
+            self.tag_ids = [Command.set(existing_tags.ids)] + [Command.create({'name': name}) for name in new_tags_names]
+        pattern = tags_and_users_group % ('(?!%s)' % ('|').join(users_to_keep) if users_to_keep else '')
+        self.display_name, _ = re.subn(pattern, '', self.display_name)
+
+    def extract_priority(self):
+        priority_group = self.get_group_pattern()['priority']
+        self.priority = "1"
+        self.display_name, _ = re.subn(priority_group, '', self.display_name)
```

Comment:
```suggestion
        self.priority = "1"
        priority_group = self.get_group_pattern()['priority']
        self.display_name, _ = re.subn(priority_group, '', self.display_name)
```
Small detail, to do one thing at a time.

Reply:
i didn't get this comment ? what's the difference ?